File: jce.Front/src/app/@core/data/models/products/Good.ts
import {Product} from './product';
import {CatalogSave} from '../catalog/catalogSave';

class Good {
  constructor(public id: number,
              public refPintel: string,
              public details: string,
              public title: string,
              public  price: number,
              public indexId: string,
              public  productTypeId: number,
              public  goodDepartmentId: number,
              public  supplierId: number,
              public  season: string,
              public  pintelSheetId: number,
              public  isBasicProduct: boolean,
              public  isBatch: boolean,
              public  isDisplayedOnJCE: boolean,
              public  isDiscountable: boolean,
              public  isEnabled: boolean,
              public  originId: number,
              public  products: Good[],
              public  createdOn: string,
              public  createdBy: string,
              public  updatedOn: string,
              public  updatedBy: string,
  ) {
  }
}

class FormGoodModel {
  constructor(public id: number,
              public refPintel: string,
              public details: string,
              public title: string,
              public  price: number,
              public indexId: string,
              public  productTypeId: number,
              public  goodDepartmentId: number,
              public  supplierId: number,
              public  season: string,
              public  pintelSheetId: number,
              public  isBasicProduct: boolean,
              public  isBatch: boolean,
              public  isDisplayedOnJCE: boolean,
              public  isDiscountable: boolean,
              public  isEnabled: boolean,
              public  originId: number,
              public  products: Good[],
              public  createdOn: string,
              public  createdBy: string,
              public  updatedOn: string,
              public  updatedBy: string) {
  }
}

export {Good, FormGoodModel}

File: jce.Server/jce.Common/Entites/JceDbContext/Good.cs
using jce.Common.Core;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;

namespace jce.Common.Entites.JceDbContext
{
    public abstract class Good : BaseEntity 
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Details { get; set; }
        public double Price { get; set; }
        public string IndexId { get; set; } 
        [Required(AllowEmptyStrings = false)]
        [MinLength(4)]
        public string RefPintel { get; set; }
        public bool IsDiscountable { get; set; }
        public bool IsEnabled { get; set; }
        public string Season { get; set; }
        public abstract void Add(Product c);
        public abstract void Remove(Product c);

        public int GoodDepartmentId { get; set; }
        public int ProductTypeId { get; set; }


        public Good()
        {
        }

        protected Good(int id, string details, int price, string indexId, string refPintel, string title, bool isEnabled, bool isDiscountable, int goodDepartment, int productType)
        {
            this.Id = id;
            this.Details = details;
            this.Price = price;
            this.IndexId = indexId;
            this.RefPintel = refPintel;
            this.Title = title;
            this.IsEnabled = isEnabled;
            this.IsDiscountable = isDiscountable;
            this.GoodDepartmentId = goodDepartment;
            this.ProductTypeId = productType;
        }
        
    }
}
